ecdsa.go

```diff
@@ -240,18 +246,18 @@ func (a *ecdsaAlgo) generatePrivateKey(seed []byte) (PrivateKey, error) {
 	}
 	defer overwrite(okm) // overwrite okm
 
-	sk := goecdsaGenerateKey(a.curve, okm)
+	sk := goecdsaMapKey(a.curve, okm)
 	return &prKeyECDSA{
 		alg:     a,
 		goPrKey: sk,
-		pubKey:  nil, // public key is not computed
+		pubKey:  nil, // public key is not constructed
```

Comment:
Now that we do the computation right away, does defering construction still bring some amount of benefit?

Reply:
The benefit is negligible, it only saves the extra structure allocation. 
The real reason is that I wanted this PR to make minimal changes till I work on supporting 1.24. There will be more internal changes at that time (to also remove the usage of deprecated functions) and I don't exclude getting back to the status where the public key computation isn't needed (non FIPS 140-3).